packages/core/src/client.ts

```diff
@@ -2903,7 +2906,8 @@ export class MedplumClient extends EventTarget {
       }
       await sleep(retryDelay);
     }
-    return this.parseResponse(resultResponse as Response, 'POST', statusUrl);
+    // @ts-expect-error resultResponse will be defined by this point, since loop will spin until a result response is received
+    return this.parseResponse(resultResponse, 'POST', statusUrl);
```

Comment:
I think this is just a cosmetic change?  This will be a merge conflict with https://github.com/medplum/medplum/pull/3937 but I think that's ok.

Reply:
Sonar labels this as a useless type assertion, using `@ts-expect-error` both makes it more clear why we can ignore this type error and makes sure that if it's no longer necessary it will throw a type error on the comment

Reply:
Oo, nice, that's a good trick.  TIL